tensorboard/default.py

```diff
@@ -86,4 +87,30 @@ def get_plugins():
 
   :rtype: list[Union[base_plugin.TBLoader, Type[base_plugin.TBPlugin]]]
   """
+
   return _PLUGINS[:]
+
+# TODO(stephanwlee): Combine with get_plugins when concept of first-party
+# plugins is undone.
+def get_dynamic_plugins():
+  """Returns a list specifying TensorBoard's dynamically loaded plugins.
+
+	A dynamic TensorBoard plugin is specified using entry_points [1]. This will be
+  the de facto mechanism to load plugins and there no longer will be a concept
+  of the first-party plugins as depicted above in `get_plugins`.
+
+  This list can be passed to the `tensorboard.program.TensorBoard` API.
+
+  Returns:
+		a list of base_plugin.TBPlugin
+
+  [1]: https://packaging.python.org/specifications/entry-points/
+  """
+  dynamic_plugins = [
+      entry_point.load()
```

Comment:
After a bit more thought, how about just returning this list as is?

```python
  return [
    entry_point.load()
    for entry_point in pkg_resources.iter_entry_points("tensorboard_plugins")
  ]
```

I’d originally wanted the symbol to be a function to accommodate plugins
that want to defer importing expensive modules as much as possible,
hence the `get_plugin()` indirection. But that’s already possible with
the `TBLoader` mechanism (e.g., `BeholderPluginLoader`), so it’s not
clear to me that further indirection is actually needed. The “name and
description” metadata is probably superfluous, and if we want to add it
later we can add optional fields/methods to `TBLoader`.

In this form, entry points would reference plugins/loaders directly:

```python
entry_points = {
    "tensorboard_plugins": [
        "widgets = tensorboard_plugin_widgets.widgets_plugin:WidgetsPlugin",
        "doodads = tensorboard_plugin_doodads.doodads_plugin:DoodadsLoader",
    ],
}
```